app/assets/javascripts/index/directions/fossgis.js

```diff
@@ -119,9 +119,9 @@ function OSRMEngine() {
           namedRoad = false;
         }
 
-        if (step.maneuver.type.match(/^exit (rotary|roundabout)$/)) {
+        if (step.maneuver.type.match(/exit (rotary|roundabout)/)) {
```

Comment:
What's the reason for removing the anchors from this, and the other matches? Is this server producing different directions that have something else before and after?

Reply:
No reason, because it is simply bitrot: the code is based on an old version, pre this commit https://github.com/openstreetmap/openstreetmap-website/commit/e9abea3dd79a11a6d756a9e3748ee6cc75911a10 of the file. I'll give bring the code up to the equivalent state.